src/v/cluster/partition_balancer_backend.cc

```diff
@@ -358,9 +362,20 @@ ss::future<> partition_balancer_backend::do_tick() {
     // status requests by default 700ms
     const auto node_responsiveness_timeout = _node_status_interval() * 7;
 
+    auto active_mode = _mode();
+
+    const auto& loaded_license = _feature_table.get_license();
+    const bool has_valid_license = loaded_license.has_value()
+                                   && !loaded_license->is_expired();
+    if (
+      active_mode == model::partition_autobalancing_mode::continuous
```

Comment:
In principle we could add a function to `config::binding` that checks whether the current value of the config is "restricted" based on the rules in `config::enterprise`. 

For no other reason than avoiding recapitulating which of `partition_autobalancing_mode` are enterprise-only. e.g. if we added a new autobalancing mode that is also restricted.

Might be too clever by half though. wdyt?

Reply:
I like it. The more we centralize the logic for when sanctions are applied (or even how, if possible), the better.